cmake/external/mkldnn.cmake

```diff
@@ -59,9 +59,13 @@ if (onnxruntime_USE_MKLDNN)
     set(MKLDNN_DLL_PATH ${MKLDNN_LIB_DIR}/${MKLDNN_SHARED_LIB})
   endif()
   set(MKLDNN_INCLUDE_DIR ${MKLDNN_INSTALL}/include)
+  set (MKLDNN_CMAKE_EXTRA_ARGS)
   if(NOT onnxruntime_BUILD_FOR_NATIVE_MACHINE)
-    set(MKLDNN_PATCH_COMMAND1 git apply ${CMAKE_SOURCE_DIR}/patches/mkldnn/platform.cmake.patch)
-    set(MKLDNN_PATCH_COMMAND2 git apply ${CMAKE_SOURCE_DIR}/patches/mkldnn/mem-patch.cmake.patch)
+    # pre-v1.0
+    list(APPEND MKLDNN_CMAKE_EXTRA_ARGS "-DARCH_OPT_FLAGS=\"\"")
+    # v1.0
+    list(APPEND MKLDNN_CMAKE_EXTRA_ARGS "-DMKLDNN_ARCH_OPT_FLAGS=\"\"")
+    set(MKLDNN_PATCH_COMMAND1 git apply ${CMAKE_SOURCE_DIR}/patches/mkldnn/mem-patch.cmake.patch)
```

Comment:
just noticed this.
why is the mem-patch only applied if NOT onnxruntime_BUILD_FOR_NATIVE_MACHINE, 
looks like an error. I think the patch should always be applied.
lines 68 and 70 should be moved outside the if check.
(I know this isn't introduced by your PR, but since you are touching this file, it's convenient to fix it?)

Reply:
No problem.
I thought it's by-design...

Reply:
we can fix it separately.